drop-storage/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+use std::str::FromStr;
+pub mod error;
+use slog::{debug, Logger};
+use sqlx::{
+    migrate::MigrateDatabase, sqlite::SqliteConnectOptions, ConnectOptions, Sqlite, SqlitePool,
+};
+
+use crate::error::Error;
+
+type Result<T> = std::result::Result<T, Error>;
+// SQLite storage wrapper
+pub struct Storage {
+    _logger: Logger,
+    #[allow(dead_code)]
+    conn: SqlitePool,
+}
+
+#[derive(Debug, strum::Display, strum::EnumString, PartialEq, Eq)]
+pub enum State {
+    #[strum(serialize = "active")]
+    Active,
+    #[strum(serialize = "canceled")]
+    Canceled,
+    #[strum(serialize = "canceled_by_peer")]
+    CanceledByPeer,
+    #[strum(serialize = "failed")]
+    Failed,
+}
+
+#[derive(Debug)]
+pub struct Transfer {
+    pub id: String,
+    pub peer: String,
+    pub state: State,
+}
+
+#[derive(Debug)]
+pub struct OutgoingTransferPath {
+    pub id: i64,
+    pub outgoing_transfer: String,
+    pub path: String,
+}
+
+#[derive(Debug)]
+pub struct OutgoingTraversedPath {
+    pub id: i64,
+    pub outgoing_path: i64,
+    pub path: String,
+    pub size: Option<i64>,
+    pub state: State,
+}
+
+#[derive(Debug)]
+pub struct IncomingTransferPath {
+    pub id: i64,
+    pub incoming_transfer: String,
+    pub path: String,
+    pub size: i64,
+    pub state: State,
+}
+
+#[derive(serde::Serialize, Debug)]
+pub enum SerializedTransferStorage {
+    Incoming {
+        id: String,
+        paths: Vec<String>,
+    },
+    Outgoing {
+        id: String,
+        path: Vec<(String, Vec<String>)>,
+    },
+}
+
+impl Storage {
+    pub async fn new(logger: Logger, path: &str) -> Result<Self> {
+        if !Sqlite::database_exists(path)
+            .await
+            .map_err(|e| error::Error::InternalError(e.to_string()))?
+        {
+            debug!(logger, "SQLite not existing. Creating database at {}", path);
+            Sqlite::create_database(path)
+                .await
+                .map_err(|e| error::Error::InternalError(e.to_string()))?;
+        }
+
+        debug!(logger, "SQLite existing: {}", path);
+
+        let mut options = SqliteConnectOptions::from_str(path)?;
+        options.log_statements(log::LevelFilter::Debug);
+
+        let conn = SqlitePool::connect_with(options)
+            .await
+            .map_err(|e| error::Error::InternalError(e.to_string()))
```

Comment:
Duplicated line?

Reply:
oh wow, really :D

Reply:
I still see it
![image](https://user-images.githubusercontent.com/41059082/233281567-18aa4791-1317-4652-99bd-a1138ac8107b.png)

Is this intentional ?